api/info/connecthandler/info_connect.go

```diff
@@ -0,0 +1,325 @@
+		return nil, err
+	}
+
+	response := &v1.GetNodeIPResponse{
+		Ip: jsonResponse.IP.String(),
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// GetNetworkID returns the numeric ID of the Avalanche network this node is connected to
+func (c *connectInfoService) GetNetworkID(
+	_ context.Context,
+	_ *connect.Request[v1.GetNetworkIDRequest],
+) (*connect.Response[v1.GetNetworkIDResponse], error) {
+	var jsonResponse info.GetNetworkIDReply
+	if err := c.Info.GetNetworkID(nil, nil, &jsonResponse); err != nil {
+		return nil, err
+	}
+
+	response := &v1.GetNetworkIDResponse{
+		NetworkId: uint32(jsonResponse.NetworkID),
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// GetNetworkName returns the name of the network
+func (c *connectInfoService) GetNetworkName(
+	_ context.Context,
+	_ *connect.Request[v1.GetNetworkNameRequest],
+) (*connect.Response[v1.GetNetworkNameResponse], error) {
+	var jsonResponse info.GetNetworkNameReply
+	if err := c.Info.GetNetworkName(nil, nil, &jsonResponse); err != nil {
+		return nil, err
+	}
+
+	response := &v1.GetNetworkNameResponse{
+		NetworkName: jsonResponse.NetworkName,
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// GetBlockchainID maps an ID string to its canonical chain ID
+func (c *connectInfoService) GetBlockchainID(
+	_ context.Context,
+	request *connect.Request[v1.GetBlockchainIDRequest],
+) (*connect.Response[v1.GetBlockchainIDResponse], error) {
+	jsonRequest := info.GetBlockchainIDArgs{
+		Alias: request.Msg.Alias,
+	}
+
+	var jsonResponse info.GetBlockchainIDReply
+	if err := c.Info.GetBlockchainID(nil, &jsonRequest, &jsonResponse); err != nil {
+		return nil, err
+	}
+
+	response := &v1.GetBlockchainIDResponse{
+		BlockchainId: jsonResponse.BlockchainID.String(),
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// Peers returns metadata (IP, nodeID, version, uptimes, subnets, etc.) for the given peer node IDs
+func (c *connectInfoService) Peers(
+	_ context.Context,
+	request *connect.Request[v1.PeersRequest],
+) (*connect.Response[v1.PeersResponse], error) {
+	nodeIDs := make([]ids.NodeID, 0, len(request.Msg.NodeIds))
+	for _, nodeIDStr := range request.Msg.NodeIds {
+		nodeID, err := ids.NodeIDFromString(nodeIDStr)
+		if err != nil {
+			return nil, connect.NewError(
+				connect.CodeInvalidArgument, fmt.Errorf("invalid nodeID %s: %w", nodeIDStr, err))
+		}
+		nodeIDs = append(nodeIDs, nodeID)
+	}
+
+	jsonRequest := info.PeersArgs{
+		NodeIDs: nodeIDs,
+	}
+
+	var jsonResponse info.PeersReply
+	if err := c.Info.Peers(nil, &jsonRequest, &jsonResponse); err != nil {
+		return nil, err
+	}
+
+	peers := make([]*v1.PeerInfo, 0, len(jsonResponse.Peers))
+	for _, peer := range jsonResponse.Peers {
+		// Convert TrackedSubnets (set.Set[ids.ID]) to []string
+		trackedSubnetsIDs := peer.TrackedSubnets.List()
+		trackedSubnets := make([]string, len(trackedSubnetsIDs))
+		for i, id := range trackedSubnetsIDs {
+			trackedSubnets[i] = id.String()
+		}
```

Comment:
In Go it's more idiomatic to use `append` instead of indexing to copy stuff over
```suggestion
		trackedSubnets := make([]string, len(trackedSubnetsIDs))
		for _, subnetID := range peer.TrackedSubnets.List() {
			trackedSubnets = append(trackedSubnets, subnetID.String())
		}
```